pulser-core/pulser/sequence/sequence.py

```diff
@@ -1938,7 +1977,19 @@ def _validate_channel(
                 "Using parametrized objects or variables to refer to channels "
                 "is not supported."
             )
-        if channel not in self._schedule:
+        if channel not in self._schedule and not (
+            self.is_parametrized()
+            and [channel]
+            == [
+                (
+                    call.args[1]
+                    if "dmm_id" not in call.kwargs
+                    else call.kwargs["dmm_id"]
+                )
+                for call in self._to_build_calls
+                if call.name == "config_slm_mask"
+            ]
```

Comment:
This makes me wonder if it's truly necessary to suspend the declaration of the DMM channel when using `config_slm_mask()`... Do you see a reason why?

Reply:
The reason why I suspended the configuration of the DMM channel when configuring the SLM mask is that otherwise at build we would end up with two configurations of the DMM

Reply:
Since `config_slm_mask` does not take parametrized arguments, I could see it be always executed as a `_call` (like we do for `declare_channel`). 
The issue I see with this is that it will change the order of the instructions and this now matters (ie what is identified as the "first pulse" will depend on when `config_slm_mask()` is called, _I think_).